source/components/IDInput/index.jsx

```diff
@@ -103,7 +116,7 @@ const IDInput = ({
               fullWidth
               value={value}
               type="text"
-              onChange={(e) => onChange(e.target.value)}
+              onChange={(e) => searchContactFromId(e)}
```

Comment:
no need for the function body here, you can pass a reference only:
`onChange ={searchContactFromId}`